homeassistant/components/modbus/binary_sensor.py

```diff
@@ -79,12 +81,19 @@ def device_class(self) -> Optional[str]:
         """Return the device class of the sensor."""
         return self._device_class
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
     def update(self):
         """Update the state of the sensor."""
-        result = self._hub.read_coils(self._slave, self._coil, 1)
         try:
+            result = self._hub.read_coils(self._slave, self._coil, 1)
+            self._available = True
             self._value = result.bits[0]
-        except AttributeError:
+        except (ConnectionException, AttributeError):
+            self._available = False
             _LOGGER.error(
```

Comment:
We should only log once when we become unavailable. The second time we should not do it or we keep spamming the log. So only log if `self._available:`